### PR TITLE
Replaces sniper bullets ex_act with bonus obj damage

### LINKED ISSUE #34761: Sniper rifles are able to hit people inside mechs.
A sniper rifle round will take off one of your limbs from inside the mech, and also deal upwards of doubled damage to the mech, oftentimes 1-2 shotting it, dealing far more than it normally does. Sniper rounds normally deal 70 brute, but against a durand (35% bullet armor) it dealt 168.8 in one hit.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
fix: Sniper rifles will no longer oneshot mechs (although they still deal high damage).
fix: Sniper rifles no longer knock down walls they hit.
/:cl:

Fixes #34761

### CODE REVIEW

**Comment by XDTM** [comment]
Tbh the bonus damage could probably be removed altogether, i'm pretty sure it was there because obj damage wasn't in the game yet.

**Comment by YPOQ** [comment]
I think it was ex_act so that the bullets could destroy walls. Similar to pulse beams/swarmer attacks.

**Comment by XDTM** [comment]
Why would a sniper rifle knock down walls, though? I can understand piercing through them, and we have specialized bullets for that, but currently they just blow the wall up and stop there.

**Comment by Rinoka** [comment]
I think it was because the ones with that function are like, on the level/design of anti material rifles lol

**Comment by Poojawa** [comment]
It'd make more sense to give them X-ray like wall penetration that chops down their viable force

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 1]
```diff
@@ -10,8 +10,10 @@
 	var/breakthings = TRUE
 
 /obj/item/projectile/bullet/p50/on_hit(atom/target, blocked = 0)
-	if((blocked != 100) && (!ismob(target) && breakthings))
-		target.ex_act(rand(1,2))
+	if((blocked != 100) && breakthings)
+		if(isobj(target))
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets/sniper.dm]: condense ifs

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**